versions/draft-2/parsers/antlr4/WdlDraft2Lexer.g4

```diff
@@ -0,0 +1,194 @@
+lexer grammar WdlDraft2Lexer;
+
+channels { COMMENTS }
+
+// Comments
+LINE_COMMENT: '#' ~[\r\n]* -> channel(COMMENTS);
+
+// Keywords
+IMPORT: 'import';
+WORKFLOW: 'workflow';
+TASK: 'task';
+SCATTER: 'scatter';
+CALL: 'call';
+IF: 'if';
+THEN: 'then';
+ELSE: 'else';
+ALIAS: 'alias';
+AS: 'as';
+In: 'in';
+INPUT: 'input';
+OUTPUT: 'output';
+PARAMETERMETA: 'parameter_meta';
+META: 'meta';
+
+HEREDOC_COMMAND: 'command' ' '* '<<<' -> pushMode(HereDocCommand);
+COMMAND: 'command' ' '* '{' -> pushMode(Command);
+
+RUNTIME: 'runtime';
+BOOLEAN: 'Boolean';
+INT: 'Int';
+FLOAT: 'Float';
+STRING: 'String';
+FILE: 'File';
+ARRAY: 'Array';
+MAP: 'Map';
+PAIR: 'Pair';
+OBJECT: 'Object';
+OBJECT_LITERAL: 'object';
+
+SEP: 'sep';
+DEFAULT: 'default';
+
+// Primitive Literals
+IntLiteral
+	: Digits
+	| SignedDigits
+	;
+FloatLiteral
+	: SignedFloatFragment
+	| FloatFragment
+	;
+BoolLiteral
+	: 'true'
+	| 'false'
+	;
+
+// Symbols
+LPAREN: '(';
+RPAREN: ')';
+LBRACE: '{' -> pushMode(DEFAULT_MODE);
+RBRACE: '}' -> popMode;
+LBRACK: '[';
+RBRACK: ']';
+ESC: '\\';
+COLON: ':';
+LT: '<';
+GT: '>';
+GTE: '>=';
+LTE: '<=';
+EQUALITY: '==';
+NOTEQUAL: '!=';
+EQUAL: '=';
+AND: '&&';
+OR: '||';
+OPTIONAL: '?';
+STAR: '*';
+PLUS: '+';
+MINUS: '-';
+DOLLAR: '$';
+COMMA: ',';
+SEMI: ';';
+DOT: '.';
+NOT: '!';
+DIVIDE: '/';
+MOD: '%';
+SQUOTE: '\'' -> pushMode(SquoteInterpolatedString);
+DQUOTE: '"' -> pushMode(DquoteInterpolatedString);
+
+WHITESPACE
+	: [ \t\r\n]+ -> channel(HIDDEN)
+	;
+
+COMMENT
+	: '#' ~[\r\n]* -> channel(HIDDEN)
+	;
+
+Identifier: CompleteIdentifier;
+
+mode SquoteInterpolatedString;
+
+SQuoteEscapedChar: '\\' . -> type(StringPart);
+SQuoteDollarString: '$'  -> type(StringPart);
+SQuoteCurlyString: '{' -> type(StringPart);
+SQuoteCommandStart: ('${') -> pushMode(DEFAULT_MODE) , type(StringCommandStart);
+SQuoteUnicodeEscape: '\\u' (HexDigit (HexDigit (HexDigit HexDigit?)?)?)? -> type(StringPart);
+EndSquote: '\'' ->  popMode, type(SQUOTE);
+StringPart: ~[${\r\n']+;
+
+mode DquoteInterpolatedString;
+
+DQuoteEscapedChar: '\\' . -> type(StringPart);
+DQuoteDollarString: '$' -> type(StringPart);
+DQUoteCurlString: '{' -> type(StringPart);
+DQuoteCommandStart: ('${') -> pushMode(DEFAULT_MODE), type(StringCommandStart);
+DQuoteUnicodeEscape: '\\u' (HexDigit (HexDigit (HexDigit HexDigit?)?)?) -> type(StringPart);
+EndDQuote: '"' ->  popMode, type(DQUOTE);
+DQuoteStringPart: ~[${\r\n"]+ -> type(StringPart);
+
+mode HereDocCommand;
+
+HereDocUnicodeEscape: '\\u' (HexDigit (HexDigit (HexDigit HexDigit?)?)?)?;
+HereDocEscapedChar: '\\' . -> type(CommandStringPart);
+HereDocCurlyString: '{' -> type(CommandStringPart);
+HereDocEscapedEnd: '\\>>>' -> type(CommandStringPart);
+EndHereDocCommand: '>>>' -> popMode, type(EndCommand);
+HereDocEscape: ( '>' | '>>' | '>>>>' '>'*) -> type(CommandStringPart);
+HereDocStringPart: ~[{>]+ -> type(CommandStringPart);
```

Comment:
I believe you are missing a token for starting a command interpolation, as well as the `$` token. It should probably have the following as well
 ```antlr
HereDoctDollarString: '$' -> type(CommandStringPart);
HereDocStringCommandStart:  ('${') -> pushMode(DEFAULT_MODE), type(StringCommandStart);
```

Reply:
Also, the `HereDocStringPart` is missing the `$` as well